src/meltano/core/migration_service.py

```diff
@@ -110,6 +110,9 @@ def upgrade(  # noqa: WPS213, WPS231 too many expression and too complex
                 fg="yellow",
                 err=True,
             )
+            raise MigrationError(
+                "Cannot upgrade the system database. It might be corrupted or was created before database migrations where introduced (v0.34.0)"
+            ) from err
```

Comment:
We'd log the same error message twice back to back with this change right? 

First via the click.secho in the line above, and then again when you raise the CliError from this MigrationError in params.py. Could probably just omit the click.secho call above.

Reply:
Thanks @pandemicsyn for checking this out. Using the `raise ... from ...` pattern I wasn't getting the "Cannot upgrade..." message in the stack trace during testing, only the root cause (`alembic.script.revision.ResolutionError: No such revision or branch 'garbage'`), so the `secho` was helpful context. However I have modified the exception raises slightly to produce both the root cause stack trace and a final helpful message via the `CliError` (in stead of relying on `secho`) 🙂